Mathlib/RingTheory/Ideal/Quotient.lean

```diff
@@ -73,28 +73,21 @@ protected def ringCon (I : Ideal R) : RingCon R :=
 #align ideal.quotient.ring_con Ideal.Quotient.ringCon
 
 instance commRing (I : Ideal R) : CommRing (R ⧸ I) :=
-  { @Submodule.Quotient.addCommGroup _ _ (id _) (id _) (id _) I,
-    inferInstanceAs (CommRing (Quotient.ringCon I).Quotient) with }
+    inferInstanceAs (CommRing (Quotient.ringCon I).Quotient)
```

Comment:
I'm not sure about this one: it looks like it's trying to make it clear that `Quotient.ringCon I` and `QuotientAddGroup.con I.toAddSubgroup` are the same relation under the hood, though maybe that's not actually effective.

As a sanity check it would probably be good to add an `example` checking that no diamond is formed with `Submodule.Quotient.addCommGroup`

Reply:
This is a good point. I’ll probably revert for this PR after testing performance. Keep it streamlined

Reply:
This `rfl`'s instantly so I'll include the example and leave as is